python/lsst/daf/butler/transfers/_yaml.py

```diff
@@ -247,6 +251,23 @@ def __init__(self, stream: IO, registry: Registry):
         self.calibAssociations: Dict[str, Dict[Timespan, List[DatasetId]]] = defaultdict(dict)
         self.refsByFileId: Dict[DatasetId, DatasetRef] = {}
         self.registry: Registry = registry
+
+        universe_version = wrapper.get("universe_version", 0)
```

Comment:
@TallJimbo would you mind taking a quick look at the changes here and in `_universe.py` to make sure they match your expectations before I merged?  I added version and namespace properties to universe and now check specifically for the old universe version when deciding to do a migration.